submissions/Eugene-CG/memory-pair-game/index.js

```diff
@@ -0,0 +1,71 @@
+const cardsList = document.querySelector(".cards__list");
+let clickRemember = [];
+class Card {
+  constructor(img) {
+    this.img = img;
+  }
+  createCard() {
+    return `
+    <li class="card__container">
+          <div class="card card-back">
+              <img src="./img/${this.img}">
+          </div>
+      </li>`;
+  }
+}
+const startGame = () => {
+  let cards = [
+    "fool.jpg",
+    "lovers.jpg",
+    "magician.jpg",
+    "priestess.jpg",
+    "star.jpg",
+    "sun.jpg",
+    "hanged.jpg",
+    "justice.jpg",
+  ];
+  cards = [...cards, ...cards];
+  shuffleCards(cards);
+  cards = createCardOjects(cards);
+  createAllCards(cards);
+};
+const shuffleCards = (cards) => cards.sort(() => Math.random() - 0.5);
+const createCardOjects = (cards) => cards.map((imgSrc) => new Card(imgSrc));
+const createAllCards = (cards) =>
+  cards.forEach((card) => (cardsList.innerHTML += card.createCard()));
```

Comment:
DOM manipulation in loops.

Adding elements to DOM from a loop is a bad practice. A browser will run reflow and repaint for every element in the loop. Instead, you can:

Use append method, which can add several elements in one operation
Create some wrapper, add your items to the wrapper and then add it to DOM. It will be one operation.
Clone current container. Add items to a container and then replace your old container with a new one. But be aware of event listeners.
Use innerHTML instead